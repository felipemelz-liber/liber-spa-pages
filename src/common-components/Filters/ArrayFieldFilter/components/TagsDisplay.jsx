import React from 'react';
import PropTypes from 'prop-types';
import { GridContainer, Grid, Tag, PlusText } from './TagsDisplay.styles';

const hasTags = tagsArray => tagsArray.length > 0;

const TagsDisplay = ({ tags, tagsPerLine, tagsThreshold }) => {
  let extraTags = [];
  let displayTags = tags;
  const columnSize = 12 / tagsPerLine;

  if (tags.length > tagsThreshold) {
    displayTags = tags.slice(0, tagsThreshold - 1);
    extraTags = tags.slice(tagsThreshold - 1);
  }
  return (
    <GridContainer spacing={2}>
      {hasTags(displayTags)
        ? displayTags.map(invoiceNumber => (
            <Grid key={invoiceNumber} columnSize={columnSize}>
              <Tag>{invoiceNumber}</Tag>
            </Grid>
          ))
        : null}
      {hasTags(extraTags) ? (
        <Grid columnSize={columnSize}>
          <PlusText>{`+ ${extraTags.length}`}</PlusText>
        </Grid>
      ) : null}
    </GridContainer>
  );
};

TagsDisplay.propTypes = {
  tags: PropTypes.arrayOf(PropTypes.string.isRequired),
  tagsThreshold: PropTypes.number.isRequired,
  tagsPerLine: PropTypes.number.isRequired,
};

TagsDisplay.defaultProps = {
  tags: [],
};

export default TagsDisplay;
