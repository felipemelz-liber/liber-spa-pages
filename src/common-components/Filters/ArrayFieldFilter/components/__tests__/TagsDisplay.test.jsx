import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import TagsDisplay from '../TagsDisplay';

describe('TagDisplay tests', () => {
  configure({ adapter: new Adapter() });

  const cases = [
    {
      tags: [],
      tagsPerLine: 6,
      tagsThreshold: 18,
    },
    {
      tags: ['a', 'b', 'c', 'd'],
      tagsPerLine: 4,
      tagsThreshold: 8,
    },
    {
      tags: ['a', 'b', 'c', 'd', 'b', 'c', 'd', 'b', 'c', 'd', 'b', 'c', 'd', 'b', 'c', 'd'],
      tagsPerLine: 5,
      tagsThreshold: 15,
    },
  ];

  cases.forEach((props, index) =>
    it(`should render correctly for case: ${index}`, () => {
      const wrapper = shallow(<TagsDisplay {...props} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    }),
  );
});
