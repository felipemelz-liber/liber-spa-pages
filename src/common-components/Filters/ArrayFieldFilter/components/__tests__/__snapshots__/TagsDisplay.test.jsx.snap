// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagDisplay tests should render correctly for case: 0 1`] = `
<styled.div
  spacing={2}
/>
`;

exports[`TagDisplay tests should render correctly for case: 1 1`] = `
<styled.div
  spacing={2}
>
  <styled.div
    columnSize={3}
    key="a"
  >
    <Styled(Tag)>
      a
    </Styled(Tag)>
  </styled.div>
  <styled.div
    columnSize={3}
    key="b"
  >
    <Styled(Tag)>
      b
    </Styled(Tag)>
  </styled.div>
  <styled.div
    columnSize={3}
    key="c"
  >
    <Styled(Tag)>
      c
    </Styled(Tag)>
  </styled.div>
  <styled.div
    columnSize={3}
    key="d"
  >
    <Styled(Tag)>
      d
    </Styled(Tag)>
  </styled.div>
</styled.div>
`;

exports[`TagDisplay tests should render correctly for case: 2 1`] = `
<styled.div
  spacing={2}
>
  <styled.div
    columnSize={2.4}
    key="a"
  >
    <Styled(Tag)>
      a
    </Styled(Tag)>
  </styled.div>
  <styled.div
    columnSize={2.4}
    key="b"
  >
    <Styled(Tag)>
      b
    </Styled(Tag)>
  </styled.div>
  <styled.div
    columnSize={2.4}
    key="c"
  >
    <Styled(Tag)>
      c
    </Styled(Tag)>
  </styled.div>
  <styled.div
    columnSize={2.4}
    key="d"
  >
    <Styled(Tag)>
      d
    </Styled(Tag)>
  </styled.div>
  <styled.div
    columnSize={2.4}
    key="b"
  >
    <Styled(Tag)>
      b
    </Styled(Tag)>
  </styled.div>
  <styled.div
    columnSize={2.4}
    key="c"
  >
    <Styled(Tag)>
      c
    </Styled(Tag)>
  </styled.div>
  <styled.div
    columnSize={2.4}
    key="d"
  >
    <Styled(Tag)>
      d
    </Styled(Tag)>
  </styled.div>
  <styled.div
    columnSize={2.4}
    key="b"
  >
    <Styled(Tag)>
      b
    </Styled(Tag)>
  </styled.div>
  <styled.div
    columnSize={2.4}
    key="c"
  >
    <Styled(Tag)>
      c
    </Styled(Tag)>
  </styled.div>
  <styled.div
    columnSize={2.4}
    key="d"
  >
    <Styled(Tag)>
      d
    </Styled(Tag)>
  </styled.div>
  <styled.div
    columnSize={2.4}
    key="b"
  >
    <Styled(Tag)>
      b
    </Styled(Tag)>
  </styled.div>
  <styled.div
    columnSize={2.4}
    key="c"
  >
    <Styled(Tag)>
      c
    </Styled(Tag)>
  </styled.div>
  <styled.div
    columnSize={2.4}
    key="d"
  >
    <Styled(Tag)>
      d
    </Styled(Tag)>
  </styled.div>
  <styled.div
    columnSize={2.4}
    key="b"
  >
    <Styled(Tag)>
      b
    </Styled(Tag)>
  </styled.div>
  <styled.div
    columnSize={2.4}
  >
    <styled.span>
      + 2
    </styled.span>
  </styled.div>
</styled.div>
`;
